domain/cloud/state/types.go

```diff
@@ -164,18 +162,32 @@ type CloudRegionDefaultValue struct {
 	Value string `db:"value"`
 }
 
-// CloudUUIDs is a slice of uuids from the database.
-type CloudUUIDs []string
+// uuids is a slice of uuids from the database.
+type uuids []string
 
-// CloudCACert represents a single row from the cloud_ca_cert table.
-type CloudCACert struct {
+// cloudCACert represents a single row from the cloud_ca_cert table.
+type cloudCACert struct {
 	// CloudUUID holds the cloud reference.
 	CloudUUID string `db:"cloud_uuid"`
 
 	// CACert holds the ca cert.
 	CACert string `db:"ca_cert"`
 }
 
+// cloudID represents only the name and UUID of a cloud
+type cloudID struct {
+	// UUID holds the cloud's unique identifier.
+	UUID string `db:"uuid"`
+	// Name holds the cloud name.
+	Name string `db:"name"`
+}
+
+// cloudName represents only the name of a cloud
```

Comment:
Ah yea, so I put it in here because it was clashing with the method parameter names, i.e we have a string argument called `cloudName` and it clashes. I've found this issue a lot for these single field structs. You have a method argument, a struct and a struct variable name, which all have to be different even though they are represent the same thing.

Reply:
I changed the type to `dbCloudName` to try and avoid this predicament. I'm keen for any other solutions though. (this was especially relevant for the `dbCloud` struct that would clash with the `cloud` package name otherwise.